flink/v1.12/flink-format/src/main/java/com/netease/arctic/flink/shuffle/LogRecordV1.java

```diff
@@ -353,20 +353,16 @@ public RowData createActualValue(Object[] objects, Type[] fieldTypes) {
       for (int i = 0; i < objects.length; i++) {
         Object obj = objects[i];
         Type type = fieldTypes[i];
-        obj = wrapIntoNullableConvert(type, obj);
+        obj = convertIfNecessary(type, obj);
         row.setField(i, obj);
       }
       return row;
     }
 
-    public Object wrapIntoNullableConvert(Type primitiveType, Object object) {
-      if (object == null) {
+    public Object convertIfNecessary(Type primitiveType, Object obj) {
```

Comment:
This is LogData.Factory super method, so I move the nullable check into the `convertIfnecessary` method.